### PR TITLE
S78.15 Fix DatasourceList

### PR DESCRIPTION
**Pull request by pvaass**

### CODE REVIEW

File: src/Klipfolio.php [hunk 1]
```diff
@@ -40,6 +40,9 @@ public function get(BaseApiConnector $connector)
 
         $response = $this->client->sendRequest('GET', $connector->getEndpoint(), ['query' => $connector->getParams()]);
 
+        if($response->hasError()) {
```
Comment (ronoudgenoeg) [inline src/Klipfolio.php]: Formatting

File: src/Object/Datasource/DatasourceList.php [hunk 2]
```diff
@@ -6,19 +6,36 @@
...
+ *
+ * @property array $datasources
  */
 class DatasourceList extends BaseApiCollection
 {
     /**
+     * BaseApiCollection constructor.
+     * @param array $data
+     */
+    public function __construct($data = []) {
```
Comment (ronoudgenoeg) [inline src/Object/Datasource/DatasourceList.php]: We do `{` after function defs on newlines

File: src/Object/Datasource/DatasourceList.php [hunk 3]
```diff
@@ -6,19 +6,36 @@
...
+        $this->data['datasources'] = [];
+        $this->setData($data);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getDatasources() {
```
Comment (ronoudgenoeg) [inline src/Object/Datasource/DatasourceList.php]: ditto

File: src/Response.php [hunk 4]
```diff
@@ -80,4 +80,19 @@ public function getLocationId()
...
+     */
+    public function hasError()
+    {
+        return array_key_exists('error_code', $this->meta) || array_key_exists('error_desc', $this->meta);
+    }
+
+    /**
+     * @return string
+     */
+    public function getError() {
```
Comment (ronoudgenoeg) [inline src/Response.php]: ditto

**Comment by pvaass** [comment]
@ronoudgenoeg fixed

**Pull request merged**